### ISSUE TITLE
Cannot install fluent-plugin-filter-parse-postfix  on Centos 6.x

### ISSUE THREAD

**Issue by ArtioDev** [fluent/fluentd#3230]
Check [CONTRIBUTING guideline](https://github.com/fluent/fluentd/blob/master/CONTRIBUTING.md) first and here is the list to help us investigate the problem.

**Describe the bug**
I cannot install fluent-plugin-filter-parse-postfix on Centos 6.x (on Centos 7.x work fine)

**To Reproduce**
 /usr/sbin/fluent-gem install fluent-plugin-filter-parse-postfix on Centos 6.x

**Expected behavior**
Install the fluent-plugin-filter-parse-postfix on Centos 6.x

**Your Environment**

- Fluentd or td-agent version: td-agent 3.8.0-0.el6
- Operating system: CentOS release 6.9 (Final)
- Kernel version: 2.6.32-573.7.1.el6.x86_64

**Your Error Log**
Building native extensions.  This could take a while...
ERROR:  Error installing fluent-plugin-filter-parse-postfix:
        ERROR: Failed to build gem native extension.

    current directory: /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/postfix_status_line-0.2.8/ext
/opt/td-agent/embedded/bin/ruby -r ./siteconf20210114-11431-jqydai.rb extconf.rb
checking for openssl/sha.h... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/opt/td-agent/embedded/bin/$(RUBY_BASE_NAME)
/opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:468:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:599:in `try_cpp'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:1107:in `block in have_header'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:957:in `block in checking_for'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:351:in `block (2 levels) in postpone'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:321:in `open'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:351:in `block in postpone'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:321:in `open'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:347:in `postpone'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:956:in `checking_for'
        from /opt/td-agent/embedded/lib/ruby/2.4.0/mkmf.rb:1106:in `have_header'
        from extconf.rb:3:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /opt/td-agent/embedded/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/postfix_status_line-0.2.8/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/postfix_status_line-0.2.8 for inspection.
Results logged to /opt/td-agent/embedded/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/postfix_status_line-0.2.8/gem_make.out

**mkmf.log:**
"gcc -o conftest -I/opt/td-agent/embedded/include/ruby-2.4.0/x86_64-linux -I/opt/td-agent/embedded/include/ruby-2.4.0/ruby/backward -I/opt/td-agent/embedded/include/ruby-2.4.0 -I. -I/opt/td-agent/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -O3 -g -pipe -I/opt/td-agent/embedded/include   -I/opt/td-agent/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -O3 -g -pipe -fPIC conftest.c  -L. -L/opt/td-agent/embedded/lib -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -Wl,-rpath,/opt/td-agent/embedded/lib -L. -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -fstack-protector -rdynamic -Wl,-export-dynamic -L/opt/td-agent/embedded/lib  -Wl,-rpath,/opt/td-agent/embedded/lib     -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -lruby  -lpthread -lrt -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

**Comment by repeatedly** [comment]
Your log contains `You have to install development tools first`.
Which packages did you install on your centos 6?

**Comment by ArtioDev** [comment]
I have installed all the "Development" tools with: **yum groupinstall "Development Tools"**

**Comment by kenhys** [comment]
I've also tried `/usr/sbin/td-agent-gem install postfix_status_line` because fluent-plugin-filter-parse-postfix depends it, it shows some more logs.

/opt/td-agent/embedded/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/postfix_status_line-0.2.8/mkmf.log

> "gcc -o conftest -I/opt/td-agent/embedded/include/ruby-2.4.0/x86_64-linux -I/opt/td-agent/embedded/include/ruby-2.4.0/ruby/backward -I/opt/td-agent/embedded/include/ruby-2.4.0 -I. -I/opt/td-agent/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -O3 -g -pipe -I/opt/td-agent/embedded/include   -I/opt/td-agent/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -O3 -g -pipe -fPIC conftest.c  -L. -L/opt/td-agent/embedded/lib -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -Wl,-rpath,/opt/td-agent/embedded/lib -L. -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -fstack-protector -rdynamic -Wl,-export-dynamic -L/opt/td-agent/embedded/lib  -Wl,-rpath,/opt/td-agent/embedded/lib     -Wl,-rpath,/opt/td-agent/embedded/lib -L/opt/td-agent/embedded/lib -lruby  -lpthread -lrt -ldl -lcrypt -lm   -lc"
In file included from /opt/td-agent/embedded/include/ruby-2.4.0/ruby.h:33,
                 from conftest.c:1:
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1381: error: wrong number of arguments specified for ‘__deprecated__’ attribute
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1390: error: wrong number of arguments specified for ‘__deprecated__’ attribute
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h: In function ‘rb_mul_size_overflow’:
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘c2’
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: ‘c2’ undeclared (first use in this function)
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: (Each undeclared identifier is reported only once
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: for each function it appears in.)
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: expected ‘)’ before ‘__int128’
/opt/td-agent/embedded/include/ruby-2.4.0/ruby/ruby.h:1615: error: expected ‘)’ before ‘__int128’
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

**Comment by repeatedly** [comment]
One possible case is gcc version mismatch.
Could you test with newer gcc?

**Comment by ArtioDev** [comment]
I've installed new version of gcc with:

sudo yum install centos-release-scl
sudo yum-config-manager --enable rhel-server-rhscl-7-rpms
sudo yum install devtoolset-7
scl enable devtoolset-7 bash

and it's WORKED. Thanks!
